public/app/core/utils/kbn.ts

```diff
@@ -1,319 +1,291 @@
-    // 25m
-    case interval < 1500000:
-      return 1200000; // 20m
-    // 45m
-    case interval < 2700000:
-      return 1800000; // 30m
-    // 1.5h
-    case interval < 5400000:
-      return 3600000; // 1h
-    // 2.5h
-    case interval < 9000000:
-      return 7200000; // 2h
-    // 4.5h
-    case interval < 16200000:
-      return 10800000; // 3h
-    // 9h
-    case interval < 32400000:
-      return 21600000; // 6h
-    // 1d
-    case interval < 86400000:
-      return 43200000; // 12h
-    // 1w
-    case interval < 604800000:
-      return 86400000; // 1d
-    // 3w
-    case interval < 1814400000:
-      return 604800000; // 1w
-    // 6w
-    case interval < 3628800000:
-      return 2592000000; // 30d
-    default:
-      return 31536000000; // 1y
-  }
-};
-
-kbn.secondsToHms = (seconds: number) => {
-  const numyears = Math.floor(seconds / 31536000);
-  if (numyears) {
-    return numyears + 'y';
-  }
-  const numdays = Math.floor((seconds % 31536000) / 86400);
-  if (numdays) {
-    return numdays + 'd';
-  }
-  const numhours = Math.floor(((seconds % 31536000) % 86400) / 3600);
-  if (numhours) {
-    return numhours + 'h';
-  }
-  const numminutes = Math.floor((((seconds % 31536000) % 86400) % 3600) / 60);
-  if (numminutes) {
-    return numminutes + 'm';
-  }
-  const numseconds = Math.floor((((seconds % 31536000) % 86400) % 3600) % 60);
-  if (numseconds) {
-    return numseconds + 's';
-  }
-  const nummilliseconds = Math.floor(seconds * 1000.0);
-  if (nummilliseconds) {
-    return nummilliseconds + 'ms';
-  }
-
-  return 'less than a millisecond'; //'just now' //or other string you like;
-};
-
-kbn.secondsToHhmmss = (seconds: number) => {
-  const strings: string[] = [];
-  const numhours = Math.floor(seconds / 3600);
-  const numminutes = Math.floor((seconds % 3600) / 60);
-  const numseconds = Math.floor((seconds % 3600) % 60);
-  numhours > 9 ? strings.push('' + numhours) : strings.push('0' + numhours);
-  numminutes > 9 ? strings.push('' + numminutes) : strings.push('0' + numminutes);
-  numseconds > 9 ? strings.push('' + numseconds) : strings.push('0' + numseconds);
-  return strings.join(':');
-};
-
-kbn.to_percent = (nr: number, outof: number) => {
-  return Math.floor((nr / outof) * 10000) / 100 + '%';
-};
-
-kbn.addslashes = (str: string) => {
-  str = str.replace(/\\/g, '\\\\');
-  str = str.replace(/\'/g, "\\'");
-  str = str.replace(/\"/g, '\\"');
-  str = str.replace(/\0/g, '\\0');
-  return str;
-};
-
-kbn.interval_regex = /(\d+(?:\.\d+)?)(ms|[Mwdhmsy])/;
-
-// histogram & trends
-kbn.intervals_in_seconds = {
-  y: 31536000,
-  M: 2592000,
-  w: 604800,
-  d: 86400,
-  h: 3600,
-  m: 60,
-  s: 1,
-  ms: 0.001,
-};
-
-kbn.calculateInterval = (range: TimeRange, resolution: number, lowLimitInterval: string[]) => {
-  let lowLimitMs = 1; // 1 millisecond default low limit
-  let intervalMs;
+const kbn = {
+  valueFormats: {} as ValueFormatterIndex,
+  interval_regex: /(\d+(?:\.\d+)?)(ms|[Mwdhmsy])/,
```

Comment:
Since you are on it, maybe we could unify the naming here to be pascalCase?

Reply:
sure, I'll do that.